metadata/src/main/java/org/apache/kafka/image/FeaturesDelta.java

```diff
@@ -66,6 +66,10 @@ public void replay(FeatureLevelRecord record) {
                         + "please ensure the metadata version is set to " + MetadataVersion.MINIMUM_VERSION + " (or higher) before "
                         + "updating the software version. The metadata version can be updated via the `kafka-features` command-line tool.", e);
             }
+        } else if (record.name().equals(KRaftVersion.FEATURE_NAME)) {
+            // Skip any feature level record for kraft.version. This has two benefits:
+            // 1. It removes from snapshots any FeatureLevelRecord for kraft.version that was incorrectly written to the log
+            // 2. Allows ApiVersions to report the correct finalized kraft.version
```

Comment:
can you add a reference to this JIRA in this comment?

Reply:
Fixed.